src/services/offer.js

```diff
@@ -227,18 +227,29 @@ class OfferService {
         return constraints.length ? { "$and": constraints } : {};
     }
 
+    isVisibleOffer(offer, hasAdminPrivileges, user) {
+        return !(offer?.isHidden && !(hasAdminPrivileges || offer.owner.toString() === user?.company?._id.toString()));
```

Comment:
It think this has the same meaning and is easier to read, and thus maintain. Please verify if it means the same... I believe it does :P

```suggestion
       return !offer?.isHidden || hasAdminPrivileges || (offer.owner.toString() === user?.company?._id.toString());
```